js/components/thumbnail.js

```diff
@@ -4,15 +4,126 @@
  * @module Thumbnail
  */
 var React = require('react'),
-    Utils = require('./utils');
+    Utils = require('./utils'),
+    ReactDOM = require('react-dom');
 
 var Thumbnail = React.createClass({
+  getInitialState: function() {
```

Comment:
Why were the initial values for the initial state chosen as such? Like, why does positionX start at -320, thumbnail width start at 80, etc?

Reply:
I got values for small thumbnails (width is 80px) (this variables (what you are asking about) are used only for 360 thumbnails), and if we get ideal sizes of images and thumbnail, we have -320 px for x-axis offset. This variables are changed after counting for real size of thumbnails and bg images.